components/browser/storage-sync-places/src/main/java/mozilla/components/browser/storage/places/Conn.kt

```diff
@@ -0,0 +1,32 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.storage.places
+
+import org.mozilla.places.PlacesConnection
+
+const val DB_NAME = "places.sqlite"
+
+/**
+ * A singleton which holds a shared reference to [PlacesConnection].
+ */
+internal object Conn {
+    @Volatile
+    private var placesConnection: PlacesConnection? = null
+
+    fun getInstance(dbPath: String) = synchronized(this) {
+        if (placesConnection != null) {
+            return@synchronized placesConnection!!
+        }
+        placesConnection = PlacesConnection(dbPath)
+        return@synchronized placesConnection!!
+    }
+
+    fun close() = synchronized(this) {
+        if (placesConnection != null) {
+            placesConnection!!.close()
+            placesConnection = null
+        }
+    }
```

Comment:
Slightly unclean: We have a `close()` here and the user of `PlacesConnection` could call `close` too; creating a situation where our connection holder will return a closed instance.